### PR TITLE
GraphQL query from CLI

### PR DESCRIPTION
**Pull request by exdis**

### CODE REVIEW

File: bin/mine [hunk 1]
```diff
@@ -16,77 +26,123 @@ function showTree(node, map, level) {
...
+        .action(() => command.run(['-h']));
+
+    const origRun = cli.Command.prototype.run;
+
+    const providers = Object.keys(dictionaries);
+
+    for (const provider of providers) {
+        const providerDict = await dictionaries[provider];
+        const { nodes } = providerDict.input;
+        const params = new Map(Object.entries(nodes).filter(([name]) => name.includes(':')));
```
Comment (lahmatiy) [inline bin/mine]:
```
node.type === 'param'
```

File: bin/mine [hunk 2]
```diff
@@ -16,77 +26,123 @@ function showTree(node, map, level) {
...
+        const miners = Object.keys(nodes).filter(node => !node.includes(':'));
+        const actors = Object.keys(providerDict.actors.nodes);
+
+        const providerCommand = command.command(provider)
+            .description(`Provider ${provider}`)
+            .action(() => providerCommand.run(['-h']));
```
Comment (lahmatiy) [inline bin/mine]:
```
this.showHelp()
```
However, due to bug in clap we need to wrap it to `console.log()` also. Would be cool to fix it in clap

File: bin/mine [hunk 3]
```diff
@@ -16,77 +26,123 @@ function showTree(node, map, level) {
...
+
+        for (const actor of actors) {
```
Comment (lahmatiy) [inline bin/mine]: It may lead to name collision with miners. Should goes after a miner or query

File: bin/mine [hunk 4]
```diff
@@ -16,77 +26,123 @@ function showTree(node, map, level) {
...
-                try {
-                    console.log(await pipeline(revertValuesNames(this.values, this.options)));
-                } catch (e) {
-                    console.error('Miner pipeline runtime error:', e);
-                    process.exit(2);
+        const queryCommand = providerCommand.command('query', '<query>')
+            .description('GraphQL query')
+            .action(async function(args) {
+                // TODO get tree from provider schema and add option to show it
+                const gql = await query(args[0]);
```
Comment (lahmatiy) [inline bin/mine]: Are you sure `await` is needed here?

File: bin/mine [hunk 5]
```diff
@@ -16,77 +26,123 @@ function showTree(node, map, level) {
...
+
+                const result = await gql.perform(provider, this.values);
+
+                if (result.errors) {
+                    result.errors.forEach(error => {
+                        console.error('Path:', error.path.join(' -> '));
```
Comment (lahmatiy) [inline bin/mine]: Should be (see in test-query.js)
```
console.error(`Query error! ${error.path ? 'Path: ' + error.path.join(' -> ') : ''}`);
```
I suppose, default query result handler should available in `lib/query`

File: bin/mine (hunk 1 above)
Comment (exdis) [reply bin/mine]: Fixed

File: bin/mine (hunk 2 above)
Comment (exdis) [reply bin/mine]: Fixed, and [also in clap](https://github.com/lahmatiy/clap/pull/12)

File: bin/mine (hunk 3 above)
Comment (exdis) [reply bin/mine]: Fixed

File: bin/mine (hunk 4 above)
Comment (exdis) [reply bin/mine]: Yep, it is unnecessary, fixed.

File: bin/mine (hunk 5 above)
Comment (exdis) [reply bin/mine]: Fixed

**Pull request merged**